jsonutil/json.go

```diff
@@ -38,3 +42,25 @@ func DecodeWithNumber(r io.Reader, value interface{}) error {
 	}
 	return nil
 }
+
+// StructFields takes a pointer to a struct and a list of exceptions,
+// and returns a list of the fields in the struct that are JSON-tagged
+// and whose tag is not in the list of exceptions.
+// The struct can be nil.
+func StructFields(s interface{}, exceptions ...string) []string {
+	st := reflect.TypeOf(s).Elem()
+	num := st.NumField()
+	fields := make([]string, 0, num)
+	for i := 0; i < num; i++ {
+		tag := st.Field(i).Tag.Get("json")
+		idx := strings.IndexByte(tag, ',')
```

Comment:
Hmm, nitpick, can you please spin this out to a separate function so that it is more obvious what is going on.